src/checkout-buttons/strategies/braintree/braintree-paypal-button-strategy.ts

```diff
@@ -33,39 +34,52 @@ export default class BraintreePaypalButtonStrategy implements CheckoutButtonStra
         }
 
         this._braintreeSDKCreator.initialize(paymentMethod.clientToken);
+        const container = `#${options.containerId}`;
+
+        this._renderButtonsData = {
+            paymentMethod,
+            paypalOptions,
+            container,
+        };
+
+        const that = this;
 
         return Promise.all([
-            this._braintreeSDKCreator.getPaypalCheckout(),
-            this._paypalScriptLoader.loadPaypal(),
+            this._braintreeSDKCreator.getPaypalCheckout(this.renderButtons, that),
+            this._braintreeSDKCreator.getPaypal(),
         ])
-            .then(([paypalCheckout, paypal]) => {
+            .then(([paypalCheckout]) => {
                 this._paypalCheckout = paypalCheckout;
+            });
+    }
 
-                const allowedSources = [];
-                const disallowedSources = [];
-
-                if (paypalOptions.allowCredit) {
-                    allowedSources.push(paypal.FUNDING.CREDIT);
-                } else {
-                    disallowedSources.push(paypal.FUNDING.CREDIT);
-                }
+    renderButtons() {
+        const allowedSources = [];
+        const disallowedSources = [];
+        const { paypalOptions, paymentMethod, container } = this._renderButtonsData as RenderButtonsData;
 
-                return paypal.Button.render({
-                    env: paymentMethod.config.testMode ? 'sandbox' : 'production',
-                    commit: paypalOptions.shouldProcessPayment ? true : false,
-                    funding: {
-                        allowed: allowedSources,
-                        disallowed: disallowedSources,
-                    },
-                    style: {
-                        shape: 'rect',
-                        label: this._offerCredit ? 'credit' : undefined,
-                        ...pick(paypalOptions.style, 'layout', 'size', 'color', 'label', 'shape', 'tagline', 'fundingicons'),
```

Comment:
Before clients can pass `label` and `shape`, now they can't. Isn't this a breaking change?

Reply:
fixed